src/Voron/Data/RawData/ActiveRawDataSmallSection.cs

```diff
@@ -176,11 +177,11 @@ public string DebugDump(RawDataSmallPageHeader* pageHeader)
                         if (oldSize->IsCompressed)
                         {
                             using var __ = Table.DecompressValue(_transaction, entryPos, size, out var buffer);
-                            OnDataMoved(prevId, newId, buffer.Ptr, buffer.Length);
+                            OnDataMoved(prevId, newId, buffer.Ptr, buffer.Length, compressed: true);
```

Comment:
Why not call this here?
` _tx.CachedDecompressedBuffersByStorageId.Remove(previousId);`

Reply:
this is only one place.
You have this one when the page is defragged and you have when the density is lower than 0.15 we merge sections.

Reply:
both which call `OnDataMoved`

Reply:
I like Karmel's approach. It's less error prone.